jgrapht-core/src/main/java/org/jgrapht/alg/matching/blossom/v5/BlossomVOptions.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * (C) Copyright 2018-2018, by Timofey Chudakov and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.matching.blossom.v5;
+
+import static org.jgrapht.alg.matching.blossom.v5.BlossomVOptions.DualUpdateStrategy.MULTIPLE_TREE_CONNECTED_COMPONENTS;
+import static org.jgrapht.alg.matching.blossom.v5.BlossomVOptions.DualUpdateStrategy.MULTIPLE_TREE_FIXED_DELTA;
+import static org.jgrapht.alg.matching.blossom.v5.BlossomVOptions.InitializationType.GREEDY;
+import static org.jgrapht.alg.matching.blossom.v5.BlossomVOptions.InitializationType.NONE;
+
+/**
+ * BlossomVOptions that define the strategies to use during the algorithm for updating duals and initializing
+ * the matching
+ * <p>
+ * According to the experimental results, the greedy initialization substantially speeds up the algorithm.
+ */
+public class BlossomVOptions {
+    /**
+     * All possible options
+     */
+    public static final BlossomVOptions[] ALL_OPTIONS = new BlossomVOptions[]{
```

Comment:
These can indeed be generated dynamically, but this will work only for extensions in the existing enums. This won't work after addition of a new enum option. Moreover, these options are now enumerated and this is the only existing enumeration. This is why it is somewhat easier to refer to a specific combination of options. It is also easier to determine which combination of options causes a specific test to fail. There were a lot of situations when the algorithm failed only with a particular combination.